src/sentry/utils/kafka.py

```diff
@@ -32,6 +33,9 @@ def get(self, key):
 
         cluster_options = settings.KAFKA_CLUSTERS[cluster_name]
         producer = self.__producers[cluster_name] = Producer(cluster_options)
+
+        atexit.register(producer.flush)
```

Comment:
This works, but it might be nice to wrap this and log from the atexit handler that it is waiting for N messages to be flushed (you can get this by calling `len(producer)`) before continuing shutdown.